gson/src/main/java/com/google/gson/Gson.java

```diff
@@ -1006,6 +1006,25 @@ private static void assertFullConsumption(Object obj, JsonReader reader) {
     return (T) fromJson(new JsonTreeReader(json), typeOfT);
   }
 
+  /**
+   * Creates a new instance of the type using either one of the {@link InstanceCreator}s
+   * {@linkplain GsonBuilder#registerTypeAdapter(Type, Object) registered with the <code>GsonBuilder</code>}
+   * or a built-in creator.
+   *
+   * <p>This method is mainly intended for {@link TypeAdapterFactory} implementations of
+   * collection-like types which want to create an empty collection instance and then add
+   * the values read from JSON to it.
+   *
+   * @param <T> type to create
+   * @param type token representing the type to create
+   * @return the created instance
+   */
+  // TODO: Specify thrown exception type; but ConstructorConstructor currently uses
+  // RuntimeException instead of specific exception
```

Comment:
Not sure if this can be resolved in a reasonable way yet.
Ideally in the future Gson's exception hierarchy should be refactored. There are exception types missing, such as for this case or when serialization fails (Gson's exceptions all seem to be related to deserialization), which currently leads to usage of `RuntimeException`.